engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/NadelFieldToService.kt

```diff
@@ -43,8 +46,10 @@ internal class NadelFieldToService(
     }
 
     private fun getService(overallField: ExecutableNormalizedField): Service {
-        if (overallField.name == Introspection.TypeNameMetaFieldDef.name) {
+        if (overallField.name == Introspection.TypeNameMetaFieldDef.name && overallField.parent != null) {
             return getService(overallField.parent)
+        } else if (overallField.name.startsWith("__")) {
```

Comment:
What happens if someone asks for `__typename` as a top level field?

Reply:
take a look at this test, let me know if it answers you question https://github.com/atlassian-labs/nadel/pull/292/files#diff-008d37ddb0e39a4a268702b2e99a0831bd625bd127cd01123b198d5a2043ea29R81

Reply:
Yup that looks good.